<!DOCTYPE html>
<html lang="zh-CN">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <link rel="stylesheet" href="">
  <title>promise</title>
</head>

<body>

  <script src="http://libs.baidu.com/jquery/2.0.0/jquery.min.js"></script>
  <script>
    function getJson({
      type = "get",
      url,
      data = null,
      dataType = "json"
    }) {

      return new Promise((resolve, reject) => {
        //异步操作的代码
        $.ajax({
          type,
          url,
          data,
          dataType,
          success: function (response) {
            //如果成功，调用resolve()将成功的数据发送出去
            resolve(response);
          },
          error: function (errText) {
            //如果失败，调用reject()将失败的消息发送出去
            reject(errText);
          }
        });
      });
    }
    // promise 实例对象上的then() 和 catch() 分别用来处理成功和失败的结果
    getJson({url:"./data/provinice.json"}).then((data) => {
      console.log(data);
      // console.log(typeof data);
    }).catch((error) => {
      console.log(error);
    })

  </script>
</body>

</html>